test/application/ladders-matchmaker.js

```diff
@@ -109,14 +116,13 @@ describe('Matchmaker', function () {
 		assert.strictEqual(matchmaker.searches.get(formatid).size, 0);
 	});
 
-	// FIXME: a race condition in battles and sockets breaks this test
-	it.skip('should create a new battle room after matchmaking', function () {
+	it('should create a new battle room after matchmaking', function () {
 		let formatid = 'gen7ou';
 		let {startBattle} = matchmaker;
 		matchmaker.startBattle = (...args) => {
 			matchmaker.startBattle = startBattle;
-			let room = matchmaker.startBattle(...args);
-			assert.ok(room instanceof Rooms.BattleRoom);
+			this.room = matchmaker.startBattle(...args);
```

Comment:
> ...then make the test asynchronous?

That breaks the `afterEach` hook

> Wait, why are you overriding matchmaker.startBattle?

Two reasons: the method does a lot of async work in it that already has broken unrelated tests, and most of the tests I have written don't need the battle to be created, just for the method to be spied on to make sure the other methods are capable of calling it.

I want to make `Matchmaker#startBattle` have as few side effects as feasible within the method itself fairly soon. It really shouldn't be doing so much I/O itself when its purposes is just to start battles.